src/context/BasketContext.tsx

```diff
@@ -21,7 +23,37 @@ export function BasketContextProvider(props: { children: JSX.Element }): JSX.Ele
 
     const addToBasket = (bi: BasketItem) => {
         const next = basket.clone();
-        next.addItem(bi);
+        const productExistence = next.items.find(
+            (item) => item.product.upc === bi.product.upc
+          );
+          if (productExistence) {
+            productExistence.quantity = productExistence.quantity + 1;
+          } else {
+            next.addItem(bi);
+          }
+        setBasket(next);
+
+        localStorage.setItem('basket', next.serialize());
```

Comment:
I see the key string `basket` all around, what do you think about doing a constant for this?